src/Collateral.sol

```diff
@@ -191,30 +229,34 @@ contract Collateral {
     }
 
     /**
-     * @dev Withdraws all thawed collateral from a receivers collateral account.
+     * @dev Withdraws all thawed collateral from a receiver's collateral account.
      * @param receiver Address of the receiver.
-     * @notice REVERT: this function will revert if the sender receiver collateral account does
-     *                 not have any thawed collateral. This function will also revert if no thawing
-     *                 period has been completed.
+     * @notice REVERT with error:
+     *               - CollateralNotThawing: There is no collateral currently thawing
+     *               - CollateralStillThawing: ThawEndTimestamp has not been reached
+     *                 for collateral currently thawing
      */
     function withdraw(address receiver) external {
         CollateralAccount storage account = collateralAccounts[msg.sender][
             receiver
         ];
-        require(account.thawEndTimestamp != 0, "No collateral thawing");
-        require(
-            account.thawEndTimestamp <= block.timestamp,
-            "Collateral still thawing"
-        );
+        if (account.thawEndTimestamp == 0) {
+            revert CollateralNotThawing();
+        }
+
+        if (account.thawEndTimestamp > block.timestamp) {
+            revert CollateralStillThawing({
+                currentTimestamp: block.timestamp,
+                thawEndTimestamp: account.thawEndTimestamp
+            });
+        }
 
         // Amount is the minimum between the amount being thawed and the actual balance
         uint256 amount = account.amountThawing > account.balance
             ? account.balance
             : account.amountThawing;
 
-        unchecked {
-            account.balance -= amount; // Reduce the balance by the withdrawn amount (no underflow risk)
-        }
+        account.balance -= amount; // Reduce the balance by the withdrawn amount (no underflow risk)
```

Comment:
Why was the unchecked removed here? Not that it makes that big of a difference but could save some gas given the previous check above.